### ISSUE TITLE
"linked panic runtime not compiled with ..." error

### ISSUE THREAD

**Issue by osa1** [rust-fuzz/afl.rs#120]
I get this error when I try running an example in [my repo](https://github.com/osa1/tiny) with afl.rs.

```
$ cargo afl build --example chat
...
error: the linked panic runtime `panic_unwind` is not compiled with this crate's panic strategy `abort`
error: aborting due to previous error
error: Could not compile `serde_derive`.
```

**Comment by frewsxcv** [comment]
a little background here. if a rust program is being fuzzed by afl and the rust program panics (via `panic_unwind`), afl does _not_ consider this a crash. because i want panics to be considered crashes, all crates compiled via `cargo afl build` are [compiled with `-C panic=abort`](https://github.com/rust-fuzz/afl.rs/blob/2cf854486dfdbe19b8a91d85e3c1a18e9ae4f154/src/bin/cargo-afl.rs#L149) which causes panics to be considered as crashes to afl.

serde_derive [has a `proc-macro` crate type](https://github.com/rust-fuzz/afl.rs/blob/2cf854486dfdbe19b8a91d85e3c1a18e9ae4f154/src/bin/cargo-afl.rs#L149). for whatever reason, rustc treats this as a `panic_unwind` panic strategy which conflicts with the `abort` panic strategy. 

a couple initial thoughts:

* it'd be good if i/we figured out _why_ proc-macro implies `panic_unwind`, this might even be a cargo/rustc bug
* it _should_ be possible to work around this by [catching panics](https://doc.rust-lang.org/nightly/std/panic/fn.catch_unwind.html) in the `afl::read_stdio_bytes` utility function and manually calling [`process:abort`](https://doc.rust-lang.org/nightly/std/process/fn.abort.html), but i think this is less elegant than just passing `-C panic=abort`

**Comment by birkenfeld** [comment]
I've also hit this issue with a proc-macro crate. Interestingly, I can use cargo-fuzz without problems.

Minimal demo project for reproduction:
[mylib.tar.gz](https://github.com/rust-fuzz/afl.rs/files/1495708/mylib.tar.gz)

**Comment by frewsxcv** [comment]
@birkenfeld @osa1 i opened a new PR that avoids changing the panic strategy in an attempt to fix this: https://github.com/rust-fuzz/afl.rs/pull/123

if either of y'all have some time, would be curious to hear if this fixes your issues:

```
cargo install --force --git https://github.com/rust-fuzz/afl.rs.git --branch frewsxcv-alt-panic
```

**Comment by osa1** [comment]
That worked. Thanks @frewsxcv .

**Comment by birkenfeld** [comment]
Works for me too!

**Comment by birkenfeld** [comment]
Although, I still wonder why it works for `cargo-fuzz`...

**Comment by frewsxcv** [comment]
fyi, i just published afl 0.3.0 which has the fix in #123
